invenio_app_ils/ui/src/invenio_app_ils/common/api/loan.js

```diff
@@ -36,6 +36,41 @@ const postAction = (
   return http.post(url, payload);
 };
 
+const buildPendingQuery = (documentPid, itemPid) => {
+  const qsDoc = documentPid ? `document_pid:${documentPid}` : '';
+  const qsItem = itemPid ? `item_pid:${itemPid}` : '';
+  const qsDocItem =
+    qsDoc && qsItem ? `(${qsDoc} OR ${qsItem})` : `${qsDoc}${qsItem}`;
+  return `${qsDocItem} AND state:PENDING`;
+};
+
+const assignItemToLoan = (itemPid, loanPid) => {
+  const newItemRef = {
+    $ref: `https://127.0.0.1:5000/api/resolver/circulation/items/${itemPid}`,
+  };
+  return http.patch(
+    `${loanURL}${loanPid}`,
+    [
+      { op: 'replace', path: '/item_pid', value: `${itemPid}` },
+      { op: 'add', path: '/item', value: newItemRef },
+    ],
+    { headers: { 'Content-Type': 'application/json-patch+json' } }
+  );
+};
+
+const fetchPendingOnDocumentItem = (
+  documentPid,
+  itemPid,
+  sortBy,
+  sortOrder
+) => {
+  const qs = buildPendingQuery(documentPid, itemPid);
+  const sort =
+    sortBy === 'transaction_date' ? `transaction_date` : `start_date`;
+  const sortByOrder = sortOrder === 'asc' ? `${sort}:asc` : `${sort}:desc`;
+  return http.get(`${loanURL}?q=${qs}&sort:${sortByOrder}`);
```

Comment:
Probably a ticket here too to use @kprzerwa refactoring? Or do we remember that this "sorting" part will disappear?

Reply:
it's not only sorting which has been refactored, also `fetchPendingOnDocumentItem` is a duplication of what 'fetchLoans' can do by passing `documentPid` and `state` params

Reply:
Issue #114